### PR TITLE
Make use of sidecar inject label across repos

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pkg/config/analysis/analyzers/util/in_mesh.go [hunk 1]
```diff
@@ -27,23 +29,30 @@ import (
...
 }
 
-func inMesh(annos map[string]string, namespace resource.Namespace, containers []v1.Container, c analysis.Context) bool {
+func inMesh(annos, labels map[string]string, namespace resource.Namespace, containers []v1.Container, c analysis.Context) bool {
 	// If pod has the sidecar container set, then, the pod is in the mesh
 	if hasIstioProxy(containers) {
 		return true
 	}
 
+	// If Pod has labels, return the injection label value
```
Comment (howardjohn) [inline pkg/config/analysis/analyzers/util/in_mesh.go:49]: The check on L55 is more robust

File: pkg/config/analysis/analyzers/util/in_mesh.go [hunk 2]
```diff
@@ -57,11 +66,11 @@ func inMesh(annos map[string]string, namespace resource.Namespace, containers []
 }
 
 // getPodSidecarInjectionStatus returns two booleans: enabled and ok.
-// enabled is true when deployment d PodSpec has either the annotation 'sidecar.istio.io/inject: "true"'
-// ok is true when the PodSpec doesn't have the 'sidecar.istio.io/inject' annotation present.
+// enabled is true when deployment d PodSpec has either the label/annotation 'sidecar.istio.io/inject: "true"'
+// ok is true when the PodSpec doesn't have the 'sidecar.istio.io/inject' label/annotation present.
 func getPodSidecarInjectionStatus(annos map[string]string) (enabled bool, ok bool) {
```
Comment (howardjohn) [inline pkg/config/analysis/analyzers/util/in_mesh.go:71]: nit: `annos` is not a great name anymore

File: pkg/config/analysis/analyzers/util/in_mesh.go [hunk 3]
```diff
@@ -57,11 +66,11 @@ func inMesh(annos map[string]string, namespace resource.Namespace, containers []
...
-	v, ok := annos[annotation.SidecarInject.Name]
-	return v == "true", ok
+	v, ok := annos[label.SidecarInject.Name]
+	return strings.EqualFold(v, "true"), ok
```
Comment (howardjohn) [inline pkg/config/analysis/analyzers/util/in_mesh.go:73]: `TRUE` is not valid

Comment (hzxuzhonghu) [reply pkg/config/analysis/analyzers/util/in_mesh.go:73]: I am not sure, actually in some other places we compare with `EqualFold`. 

Can you confirm?

Comment (hzxuzhonghu) [reply pkg/config/analysis/analyzers/util/in_mesh.go:73]: Also ParseBool is used

**Comment by howardjohn** [comment]
The label selector in the k8s webhook is definitely not EqualFold so there
may be some inconsistencies between different areas.

File: pkg/config/analysis/analyzers/util/in_mesh.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/config/analysis/analyzers/util/in_mesh.go:49]: Confused, shouldn't labels be respected over annotations

**Comment by hzxuzhonghu** [comment]
@istio/wg-user-experience-maintainers

File: pkg/config/analysis/analyzers/util/in_mesh.go (hunk 1 above)
Comment (howardjohn) [reply pkg/config/analysis/analyzers/util/in_mesh.go:47]: I misunderstood

**Review by howardjohn — APPROVED** [review]

**Comment by hzxuzhonghu** [comment]
kindly ping @istio/wg-user-experience-maintainers

**Review by hanxiaop — APPROVED** [review]
LGTM

**Pull request merged**